deepchecks/tabular/metric_utils/metrics.py

```diff
@@ -0,0 +1,279 @@
+#
+"""Utils module containing utilities for checks working with metrics."""
+import typing as t
+from numbers import Number
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.metrics import f1_score, get_scorer, make_scorer, precision_score, recall_score
+from sklearn.metrics._scorer import _BaseScorer, _ProbaScorer
+
+from deepchecks import tabular  # pylint: disable=unused-import; it is used for type annotations
+from deepchecks.core import errors
+from deepchecks.tabular.utils.task_type import TaskType
+from deepchecks.utils.logger import get_logger
+from deepchecks.utils.simple_models import PerfectModel
+from deepchecks.utils.strings import is_string_column
+from deepchecks.utils.typing import BasicModel, ClassificationModel
+
+__all__ = [
+    'task_type_check',
+    'DEFAULT_SCORERS_DICT',
+    'DEFAULT_REGRESSION_SCORERS',
+    'DEFAULT_BINARY_SCORERS',
+    'DEFAULT_MULTICLASS_SCORERS',
+    'MULTICLASS_SCORERS_NON_AVERAGE',
+    'DeepcheckScorer',
+    'init_validate_scorers',
+    'get_default_scorers'
+]
+
+
+DEFAULT_BINARY_SCORERS = {
+    'Accuracy': 'accuracy',
+    'Precision': make_scorer(precision_score, zero_division=0),
+    'Recall': make_scorer(recall_score, zero_division=0)
+}
+
+DEFAULT_MULTICLASS_SCORERS = {
+    'Accuracy': 'accuracy',
+    'Precision - Macro Average': make_scorer(precision_score, average='macro', zero_division=0),
+    'Recall - Macro Average': make_scorer(recall_score, average='macro', zero_division=0),
+}
+
+MULTICLASS_SCORERS_NON_AVERAGE = {
+    'F1': 'f1_per_class',
+    'Precision': 'precision_per_class',
+    'Recall': 'recall_per_class',
+}
+
+DEFAULT_REGRESSION_SCORERS = {
+    'Neg RMSE': 'neg_root_mean_squared_error',
+    'Neg MAE': 'neg_mean_absolute_error',
+    'R2': 'r2',
+}
+
+DEFAULT_SCORERS_DICT = {
+    TaskType.BINARY: DEFAULT_BINARY_SCORERS,
+    TaskType.MULTICLASS: DEFAULT_MULTICLASS_SCORERS,
+    TaskType.REGRESSION: DEFAULT_REGRESSION_SCORERS
+}
+
+
+_func_dict = {
+    'precision_per_class': make_scorer(precision_score, average=None, zero_division=0),
+    'recall_per_class': make_scorer(recall_score, average=None, zero_division=0),
+    'f1_per_class': make_scorer(f1_score, average=None, zero_division=0)
+}
+
+
+class DeepcheckScorer:
+    """Encapsulate scorer function with extra methods.
+
+    Scorer functions are functions used to compute various performance metrics, using the model and data as inputs,
+    rather than the labels and predictions. Scorers are callables with the signature scorer(model, features, y_true).
+    Additional data on scorer functions can be found at https://scikit-learn.org/stable/modules/model_evaluation.html.
+
+    Parameters
+    ----------
+    scorer : t.Union[str, t.Callable]
+        sklearn scorer name or callable
+    name : str
+        scorer name
+    """
+
+    def __init__(self, scorer: t.Union[str, t.Callable], name: str):
```

Comment:
Additional request - add a class field indicating whether the scorer returns a float or a list of float + a public method is_scalar_scorer

Reply:
not now